File: DataStructure/DifferentialArray/leetcode2406/solve.go
// type Pair struct {

// 	start, end int

// }

// func minGroups(intervals [][]int) int {

// 	diff := make(map[int]int)

// 	for _, interval := range intervals {

// 		diff[interval[0]]++

// 		diff[interval[1]+1]--

// 	}

// 	// 将 map 的键排序以进行遍历

// 	var keys []int

// 	for k := range diff {

// 		keys = append(keys, k)

// 	}

// 	sort.Ints(keys)

// 	ans, sum := 0, 0

// 	for _, k := range keys {

// 		sum += diff[k]

// 		if sum > ans {

// 			ans = sum

// 		}

// 	}

// 	return ans

// }

File: DataStructure/DifferentialArray/leetcode2406/solve.cpp
// class Solution {
// public:
//     int minGroups(vector<vector<int>>& intervals) {
//         map<int,int>diff;
//         for(const auto&interval:intervals)
//             ++diff[interval[0]],--diff[interval[1]+1];
//         int ans=0,sum=0;
//         for(auto[_,d]:diff)
//         {
//             ans=max(ans,sum+=d);
//         }
//         return ans;
//     }
// };